bundles/org.openhab.binding.sunsynk/src/main/java/org/openhab/binding/sunsynk/internal/api/AccountController.java

```diff
@@ -0,0 +1,163 @@
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonSyntaxException;
+
+/**
+ * The {@link AccountController} is the internal class for a Sunsynk Connect
+ * Account.
+ * 
+ * @author Lee Charlton - Initial contribution
+ */
+
+@NonNullByDefault
+public class AccountController {
+    private final Logger logger = LoggerFactory.getLogger(AccountController.class);
+
+    private @Nullable Client sunAccount = new Client();
+
+    public AccountController() {
+    }
+
+    public void authenticate(String username, String password)
+            throws SunSynkAuthenticateException, SunSynkTokenException {
+        String payload = makeLoginBody(username, password);
+        sendHttp(payload);
+    }
+
+    public void refreshAccount(String username) throws SunSynkAuthenticateException, SunSynkTokenException {
+        if (sunAccount == null) {
+            throw new SunSynkAuthenticateException(
+                    "Sun Synk Account could not be authenticated: Try re-enabling account");
+        } else if (!sunAccount.dataIsNotNull()) {
+            throw new SunSynkAuthenticateException(
+                    "Sun Synk Account could not be authenticated: Try re-enabling account");
+        }
+        Long expires_in = this.sunAccount.getExpiresIn();
+        Long issued_at = this.sunAccount.getIssuedAt();
+        if ((issued_at + expires_in) - Instant.now().getEpochSecond() > 30) { // > 30 seconds
+            logger.debug("Account configuration token not expired.");
+            return;
+        }
+        logger.debug("Account configuration token expired : {}", this.sunAccount.getData().toString());
+        String payload = makeRefreshBody(username, this.sunAccount.getRefreshTokenString());
+        sendHttp(payload);
+    }
+
+    private void sendHttp(String payload) throws SunSynkAuthenticateException, SunSynkTokenException {
+        Gson gson = new Gson();
+        String response = "";
+        String httpsURL = makeLoginURL("oauth/token");
+        Properties headers = new Properties();
+        headers.setProperty("Accept", "application/json");
+        headers.setProperty("Requester", "www.openhab.org"); // optional
+        InputStream stream = new ByteArrayInputStream(payload.getBytes(StandardCharsets.UTF_8));
+        try {
+            response = HttpUtil.executeUrl("POST", httpsURL, headers, stream, "application/json", 2000);
+            this.sunAccount = gson.fromJson(response, Client.class);
+        } catch (IOException | JsonSyntaxException e) {
+            throw new SunSynkAuthenticateException("Sun Synk Account could not be authenticated", e);
+        }
+        if (this.sunAccount == null) {
+            throw new SunSynkAuthenticateException("Sun Synk Account failed to get a response.");
+        }
+        if (this.sunAccount.getCode() == 102) {
+            logger.info("Sun Synk Account could not be authenticated: {}.", this.sunAccount.getMsg());
```

Comment:
```suggestion
            logger.debug("Sun Synk Account could not be authenticated: {}.", this.sunAccount.getMsg());
```
As the `SunSynkAuthenticateException` is routed to a Thing status, this should not be logged to a default level.